### PR TITLE
Fix relative path test on Windows

### PR DESCRIPTION
**Pull request by nywilken**
This changes fixes the one test for Windows, but I am not entirely
confident that the test passing means that it will work 100% of the time
on Windows. I am still wrapping my head around how the submission stuff
should play out.

### CODE REVIEW

**Comment by kytrinyx** [comment]
> I am still wrapping my head around how the submission stuff should play out.

You and me both!

File: cmd/submit_relative_path_windows_test.go [hunk 1]
```diff
@@ -55,5 +56,5 @@ func TestSubmitRelativePath(t *testing.T) {
 	assert.NoError(t, err)
 
 	assert.Equal(t, 1, len(submittedFiles))
-	assert.Equal(t, "This is a file.", submittedFiles["/file.txt"])
+	assert.Equal(t, "This is a file.", submittedFiles["\\file.txt"])
```
Comment (kytrinyx) [inline cmd/submit_relative_path_windows_test.go]: OMG. Of course.

This should be `string(os.PathSeparator) + "file.txt"` or something like that. I wonder if that will automatically escape the string.

File: cmd/submit_relative_path_windows_test.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ func TestSubmitRelativePath(t *testing.T) {
 
 	tmpDir, err := ioutil.TempDir("", "relative-path")
 	assert.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
```
Comment (kytrinyx) [inline cmd/submit_relative_path_windows_test.go]: Nice. I need to add this logic everywhere.

File: cmd/submit_relative_path_windows_test.go (hunk 1 above)
Comment (nywilken) [reply cmd/submit_relative_path_windows_test.go]: I don't think it will escape, but escaping is not needed as the key actually is `\file.txt`. Changing to `string(os.PathSeparator) + "file.txt"` works as expected. However, I have a question why does the filename need to have the path separator? Should it not just be the filename?

Comment (kytrinyx) [reply cmd/submit_relative_path_windows_test.go]: It's what the exercism API sends back, relative to the workspace directory, including the path separator. We could have done it either way, I don't think we had any reason to do one over the other.

Comment (kytrinyx) [reply cmd/submit_relative_path_windows_test.go]: s/sends back/expects/ (sends back on the download)

Comment (kytrinyx) [reply cmd/submit_relative_path_windows_test.go]: Wait. uhm.... is it just setup in the test that does this? In that case I think you're right, we should just remove it.

**Comment by kytrinyx** [comment]
I made a tweak to the branches on this repository.

**TL;DR: You need to edit this PR to be against `master` instead of `nextercism`**

- I renamed the old master to `v2.x` (I might just make this a tag, but for now it's a branch)
- I pushed a copy of `nextercism` up to `master`

Once we've made sure everything looks right we can delete the nextercism branch.

File: cmd/submit_relative_path_windows_test.go (hunk 1 above)
Comment (kytrinyx) [reply cmd/submit_relative_path_windows_test.go]: I think the most reasonable thing right now is to keep the leading slash, since there are so many moving parts. It could be worth reconsidering down the road, as whether we change this now or later, we still have to be able to take into account the exercises that _do_ have the leading slash.

Comment (nywilken) [reply cmd/submit_relative_path_windows_test.go]: I agree with you on this one; especially since I have not been able to give a lot of time. I would hate to start a larger thing right now and not be able to follow it all the way through. 

If you are okay with it I will merge this in.

Comment (kytrinyx) [reply cmd/submit_relative_path_windows_test.go]: Yepp, I think that's fine. I'll go ahead and merge.
